shuler_lab_to_nwb/fiberphotometry_datainterface/metadata.yaml

```diff
@@ -0,0 +1,105 @@
+NWBFile:
+  session_description: Fiber photometry recordings.
+  session_start_time: "2020-01-01T12:30:00-08:00"
+  institution: Johns Hopkins University School of Medicine
+  lab: Shuler Lab
+  experimenter:
+    - Zhang, Shichen
+Subject:
+  species: Mus musculus
+  age: 20W
+  sex: U
+FiberPhotometry:
+  ExcitationSourcesTable:
+    description: The excitation sources table.
+    items:
+      - peak_wavelength: 475.0 # in nanometers
+        source_type: LED # LED or laser
+      - peak_wavelength: 560.0 # in nanometers
+        source_type: LED # LED or laser
+  PhotodetectorsTable:
+    description: The metadata for the photodetector.
+    items:
+      - peak_wavelength: 510.0 # in nanometers
+        type: CMOS camera
+  FluorophoresTable:
+    description: The metadata for the fluorophores. The coordinates are in unit meters relative to Bregma (AP, ML, DV).
+    items:
+      - label: GFP
+        location: Right hemisphere.
+        coordinates:
+          - 0.0008
+          - 0.0015
+          - 0.00233
+      - label: dsRed
+        location: Right hemisphere.
+        coordinates:
+          - 0.0008
+          - 0.0015
+          - 0.00233
+      - label: GFP
+        location: Left hemisphere.
+        coordinates:
+          - 0.0008
+          - -0.0015
+          - 0.00233
+      - label: dsRed
+        location: Left hemisphere.
+        coordinates:
+          - 0.0008
+          - -0.0015
+          - 0.00233
+  FibersTable:
+    description: The metadata for the optical fiber.
+    items:
+      - location: Right hemisphere.
+        notes: The optical fiber was implanted at 0.0008 m (AP), 0.0015 m (ML), and 0.00233 m (DV) relative to Bregma.
+        photodetector: 0
+        excitation_source:
+          - 0
+          - 1
```

Comment:
How would you feel about swapping to using `str`-valued object (`dict`) keys instead of lists of integers? Would it help identify/track the linkage?

Reply:
@CodyCBakerPhD I don't understand what you're proposing here, care to explain?

Reply:
Instead of relying on the order in the YAML list (the line the `-` character occurs on), have these be encoded as key-value pairings [such as those seen here](https://github.com/catalystneuro/datta-lab-to-nwb/blob/main/src/datta_lab_to_nwb/markowitz_gillis_nature_2023/markowitz_gillis_nature_2023_metadata.yaml#L78-L84)

Of course for identity mappings this might seem strange and trivial; it really becomes most obvious and beneficial for non-identity mappings

Reply:
ah I see what you mean, but I don’t see the advantage of making this in this case, where all we need to link is an unordered list of indexes